File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(wants, number, discount) {
    let result = 0;
    for(let i = 0 ; i < discount.length - 9; i++){
        const window = discount.slice(i, i+10);
        const counter = {};
        
        for(const item of window){
            counter[item] = (counter[item] || 0) + 1
        }
        
        let success = true;
        for(let j = 0 ; j < wants.length ; j++){
            const want = wants[j];
            const num = number[j];
            
            if(!counter[want] || counter[want] < num){
                success = false;
                break;
            }
        }
        if(success) result++;
    }
    return result;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter
def solution(wants, number, discount):
    result = 0
    total_days = len(discount)
    
    for i in range(total_days - 9):
        want_cnt = Counter(discount[i : i + 10])
        
        success = True
        for want, num in zip(wants, number):
            if want_cnt[want] < num:
                success = False
                break
                
        if success:
            result += 1
            
    return result